Samples/NetworkMediaStreamer/RTPMediaStreamer/src/RTPStreamSink.cpp

```diff
@@ -0,0 +1,433 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+#include <pch.h>
+
+TxContext::TxContext(std::string destination, winrt::PacketHandler packetHandler /*= nullptr*/)
+    : m_u64StartTime(0)
+    , m_packetHandler(packetHandler)
+    , m_ssrc(0)
+    , m_RtpSocket(INVALID_SOCKET)
+    , m_RtcpSocket(INVALID_SOCKET)
+    , m_localRTPPort(0)
+    , m_localRTCPPort(0)
+    , m_remotePort(0)
+    , m_SequenceNumber(0)
+{
+    memset(&m_remoteAddr, 0, sizeof(m_remoteAddr));
+    std::string param = "ssrc=";
+    auto sep2 = destination.find(param);
+    if (sep2 != std::string::npos)
+    {
+        auto sep3 = destination.find("&", sep2);
+        m_ssrc = std::stoi(destination.substr(sep2 + param.size(), sep3 - sep2));
+    }
+    else
+    {
+        m_ssrc = 0;
+    }
+
+    if (!packetHandler)
+    {
+       param = "localrtpport=";
+        auto sep1 = destination.find(":");
+        auto ipaddr = destination.substr(0, sep1);
+        sep2 = destination.find(param);
+        m_remotePort = std::stoi(destination.substr(sep1 + 1, sep2 - sep1 - 1));
+        if (sep2 != std::string::npos)
+        {
+            auto sep3 = destination.find("&", sep2);
+            m_localRTPPort = std::stoi(destination.substr(sep2 + param.size(), sep3 - sep2));
+        }
+        else
+        {
+            m_localRTPPort = 6970;
+        }
+        //else
+        {
+            sockaddr_in Server;
+            Server.sin_family = AF_INET;
+            Server.sin_addr.s_addr = INADDR_ANY;
+            for (u_short P = m_localRTPPort; P < 0xFFFE; P += 2)
+            {
+                m_RtpSocket = socket(AF_INET, SOCK_DGRAM, 0);
+                Server.sin_port = htons(P);
+                if (bind(m_RtpSocket, (sockaddr*)&Server, sizeof(Server)) == 0)
+                {   // Rtp socket was bound successfully. Lets try to bind the consecutive Rtsp socket
+                    m_RtcpSocket = socket(AF_INET, SOCK_DGRAM, 0);
+                    Server.sin_port = htons(P + 1);
+                    if (bind(m_RtcpSocket, (sockaddr*)&Server, sizeof(Server)) == 0)
+                    {
+                        m_localRTPPort = P;
+                        m_localRTCPPort = P + 1;
+                        break;
+                    }
+                    else
+                    {
+                        closesocket(m_RtpSocket);

```

Comment:
>m_RtpSocket [](start = 36, length = 11)

should this be set to invalid value on close, to know if you should cleanup on destructor